plugin/core/sessions.py

```diff
@@ -1032,6 +1038,21 @@ def m_client_unregisterCapability(self, params: Any, request_id: Any) -> None:
                         sv.on_capability_removed_async(registration_id, discarded)
         self.send_response(Response(request_id, None))
 
+    def m_window_showDocument(self, params: Any, request_id: Any) -> None:
+        """handles the window/showDocument request"""
+        uri = params.get("uri")
+
+        def success(b: bool) -> None:
+            self.send_response(Response(request_id, {"success": b}))
+
+        if params.get("external"):
+            success(open_externally(uri, bool(params.get("takeFocus"))))
+        else:
+            # TODO: ST API does not allow us to say "do not focus this new view"
+            filename = uri_to_filename(uri)
+            selection = params.get("selection")
+            open_file_and_center_async(self.window, filename, selection).then(lambda _: success(True))
```

Comment:
Should we take the `open_file_*` take a region right away? Feels weird to pass this poorly typed Dict.

Reply:
I pondered on this for a while and I think we should keep using a loose dict. The larger issue is that Range is such a weird class. We have sublime.Region which can already do everything that Range can do. Perhaps we can remove the Range class in the future?